lib/ch/connection.ex

```diff
@@ -21,7 +21,7 @@ defmodule Ch.Connection do
           case DBConnection.Query.decode(handshake, responses, _opts = []) do
             %Result{rows: [[1, version]]} ->
               conn =
-                if version >= "24.10" do
+                if parse_version(version) >= parse_version("24.10") do
```

Comment:
I couldn't make `Version.compare` work so I added a custom function to decode versions.